src/hotspot/cpu/riscv/nativeInst_riscv.hpp

```diff
@@ -208,18 +208,18 @@ class NativeInstruction {
  protected:
   address addr_at(int offset) const    { return address(this) + offset; }
 
-  jint int_at(int offset) const        { return *(jint*) addr_at(offset); }
-  juint uint_at(int offset) const      { return *(juint*) addr_at(offset); }
+  jint int_at(int offset) const        { return (jint)Bytes::get_native_u4(addr_at(offset)); }
+  juint uint_at(int offset) const      { return Bytes::get_native_u4(addr_at(offset)); }
 
-  address ptr_at(int offset) const     { return *(address*) addr_at(offset); }
+  address ptr_at(int offset) const     { return (address)Bytes::get_native_u8(addr_at(offset)); }
 
-  oop  oop_at (int offset) const       { return *(oop*) addr_at(offset); }
+  oop  oop_at (int offset) const       { return cast_to_oop(Bytes::get_native_u8(addr_at(offset))); }
 
 
-  void set_int_at(int offset, jint  i)        { *(jint*)addr_at(offset) = i; }
-  void set_uint_at(int offset, jint  i)       { *(juint*)addr_at(offset) = i; }
-  void set_ptr_at (int offset, address  ptr)  { *(address*) addr_at(offset) = ptr; }
-  void set_oop_at (int offset, oop  o)        { *(oop*) addr_at(offset) = o; }
+  void set_int_at(int offset, jint  i)        { Bytes::put_native_u4(addr_at(offset), i); }
+  void set_uint_at(int offset, jint  i)       { Bytes::put_native_u4(addr_at(offset), i); }
+  void set_ptr_at (int offset, address  ptr)  { Bytes::put_native_u8(addr_at(offset), (u8)ptr); }
+  void set_oop_at (int offset, oop  o)        { Bytes::put_native_u8(addr_at(offset), cast_from_oop<u8>(o)); }
```

Comment:
I see there are two spaces between type and name for the second parameter. We should remove one.